src/Tribe/Integrations/WPML/Views/V2/Filters.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Manages integration between WPML and Views v2 of The Events Calendar.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Integrations\WPML\Views\V2
+ */
+
+namespace Tribe\Events\Integrations\WPML\Views\V2;
+
+/**
+ * Class Filters
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Integrations\WPML\Views\V2
+ */
+class Filters {
+	/**
+	 * Translates the View URL.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $url The original View URL.
+	 *
+	 * @return string The translated View URL.
+	 */
+	public static function translate_view_url( $url ) {
+		$lang = static::get_request_lang();
+		if ( false === $lang ) {
+			return $url;
+		}
+
+		return apply_filters( 'wpml_permalink', $url, $lang );
+	}
+
+	/**
+	 * Returns the current request language, read from the request cookie.
+	 *
+	 * @since TBD
+	 *
+	 * @return string|false Either the request language, e.g. `fr`, or `false` to indicate the language could not be
+	 *                      parsed from the request context or that the request context is the current one and there
+	 *                      is no need to translate URLs.
+	 */
+	protected static function get_request_lang() {
+		if ( ! (
+			function_exists( 'icl_object_id' )
+			&& isset( $_COOKIE['wp-wpml_current_language'] )
+		) ) {
+			return false;
+		}
+
+		$request_lang = tribe_get_request_var( 'wp-wpml_current_language', false );
+
+		if ( $request_lang === ICL_LANGUAGE_CODE ) {
+			return false;
+		}
+
+		return $request_lang;
+	}
+
+	/**
+	 * Translates the URls contained in the View template variables.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $template_vars The original View template variables.
+	 *
+	 * @return array<string,mixed> The View template variables, with the URLs there contained translated, if required.
+	 */
+	public static function translate_template_vars_urls( $template_vars = [] ) {
+		if ( ! is_array( $template_vars ) ) {
+			return $template_vars;
+		}
+
+		$url_keys = array_filter( $template_vars, static function ( $value, $key ) {
+			return str_contains( $key, 'url' ) && (bool) filter_var( $value, FILTER_VALIDATE_URL );
+		}, ARRAY_FILTER_USE_BOTH );
+
```

Comment:
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 72fbbf165cea978485699af4d7cb0c44</sub>